diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2fe2f21..c56df49f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,7 @@ option(ENABLE_VOICE                  "Enable voice support"
 if (NOT ONLY_SLEEPY_DISCORD)
 	option(AUTO_DOWNLOAD_LIBRARY         "Automatically download sleepy discord standard config dependencies" ON )
 	#option(SLEEPY_FIND_DEPS              "Find Sleepy Discord dependencies using find_ "                      OFF)
+	option(USE_VCPKG_LIBS                "Use VCPKG for libraries"                                            OFF)
 	option(USE_CPR                       "Use CPR http library"                                               ON )
 	option(USE_WEBSOCKETPP               "Use websocketpp websockets library"                                 ON )
 	option(USE_UWEBSOCKETS               "Use uWebsockets websockets library"                                 OFF)
@@ -21,6 +22,11 @@ endif()
 #Define a variable to use to check if this file has been executed
 set(SLEEPY_DISCORD_CMAKE ON)
 
+# AUTO_DOWNLOAD_LIBRARY and USE_VCPKG_LIBS are mutually exclusive
+if (AUTO_DOWNLOAD_LIBRARY AND USE_VCPKG_LIBS)
+	message(FATAL_ERROR "Can't auto download and use vcpkg libraries at the same time")
+endif()
+
 #set variables based on other variables
 if(BUILD_EXAMPLES)
 	set(SLEEPY_DISCORD_BUILD_EXAMPLES ON)
@@ -139,6 +145,8 @@ if(AUTO_DOWNLOAD_LIBRARY)
 			UPDATE_DISCONNECTED 1
 		)
 	endif()
+elseif(USE_VCPKG_LIBS)
+	# nothing to do here.
 else()
 	if(USE_CPR)
 		set(cpr_SOURCE_DIR ${PROJECT_SOURCE_DIR}/deps/cpr)
@@ -171,49 +179,51 @@ if(USE_CPR)
 	set(CURL_STATICLIB ON CACHE BOOL "")
 	set(CURL_ZLIB OFF CACHE BOOL "")
 
-	#set what ssl library to use for curl based on OS
-	#this fixes
-	if(NOT DEFINED CMAKE_USE_OPENSSL)
-		if(WIN32)
-			set(CMAKE_USE_WINSSL ON CACHE BOOL "")
-			set(USE_WINSSL ON CACHE BOOL "")
-			set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
-			set(USE_OPENSSL OFF CACHE BOOL "")
-		elseif(APPLE)
-			set(CMAKE_USE_DARWINSSL ON CACHE BOOL "")
-			set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
-		elseif(UNIX)
-			set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
+	if(NOT USE_VCPKG_LIBS)
+		#set what ssl library to use for curl based on OS
+		#this fixes
+		if(NOT DEFINED CMAKE_USE_OPENSSL)
+			if(WIN32)
+				set(CMAKE_USE_WINSSL ON CACHE BOOL "")
+				set(USE_WINSSL ON CACHE BOOL "")
+				set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
+				set(USE_OPENSSL OFF CACHE BOOL "")
+			elseif(APPLE)
+				set(CMAKE_USE_DARWINSSL ON CACHE BOOL "")
+				set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
+			elseif(UNIX)
+				set(CMAKE_USE_OPENSSL OFF CACHE BOOL "")
+			endif()
 		endif()
-	endif()
 
-	if(NOT DEFINED BUILD_SHARED_LIBS AND WIN32)
-		set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
-	endif()
+		if(NOT DEFINED BUILD_SHARED_LIBS AND WIN32)
+			set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
+		endif()
 
-	#using cmake to configure curl has issues
-	#so use system curl
-	set(USE_SYSTEM_CURL ON CACHE BOOL "")
+		#using cmake to configure curl has issues
+		#so use system curl
+		set(USE_SYSTEM_CURL ON CACHE BOOL "")
 
-	add_subdirectory(deps/cpr)
+		add_subdirectory(deps/cpr)
+	endif()
 endif()
 
-if(USE_WEBSOCKETPP)
+if(USE_WEBSOCKETPP AND NOT USE_VCPKG_LIBS)
 	set(THREADS_PREFER_PTHREAD_FLAG ON)
 	find_package(Threads REQUIRED)
 	add_subdirectory(deps/websocketpp)
 endif()
 
-if(USE_LIBOPUS)
+if(USE_LIBOPUS AND NOT USE_VCPKG_LIBS)
 	add_subdirectory(deps/opus)
 endif()
 
-if(USE_ZLIB_NG)
+if(USE_ZLIB_NG AND NOT USE_VCPKG_LIBS)
 	add_subdirectory(deps/zlib-ng)
 endif()
 
 # Get Version Info
-if(Git_FOUND)
+if(Git_FOUND AND NOT USE_VCPKG_LIBS)
 	execute_process(
 		WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} 
 		COMMAND ${GIT_EXECUTABLE} rev-list --count HEAD
@@ -263,10 +273,18 @@ else()
 endif()
 
 if(NOT GIT_INFO_FOUND)
-	configure_file(
-		"${PROJECT_SOURCE_DIR}/include/sleepy_discord/version-non.h.in"
-		"${PROJECT_SOURCE_DIR}/include/sleepy_discord/version.h"
-	)
+	if (NOT DEFINED SLEEPY_DISCORD_VERSION_BUILD
+	    OR NOT DEFINED SLEEPY_DISCORD_VERSION_BRANCH
+	    OR NOT DEFINED SLEEPY_DISCORD_VERSION_HASH
+	    OR NOT DEFINED SLEEPY_DISCORD_VERSION_BRANCH
+	    OR NOT DEFINED SLEEPY_DISCORD_VERSION_DESCRIPTION)
+		set(NO_VERSION_INFO ON)
+	else()
+		configure_file(
+			"${PROJECT_SOURCE_DIR}/include/sleepy_discord/version-non.h.in"
+			"${PROJECT_SOURCE_DIR}/include/sleepy_discord/version.h"
+		)
+	endif()
 endif()
 
 # Add Subdirectories
