diff --git a/include/cpr/body.h b/include/cpr/body.h
index 04dc1c9..e6b5089 100644
--- a/include/cpr/body.h
+++ b/include/cpr/body.h
@@ -29,10 +29,10 @@ class Body : public StringHolder<Body> {
         std::ifstream is(file.filepath, std::ifstream::binary);
         if (is) {
             is.seekg(0, is.end);
-            int length = is.tellg();
+            const std::streampos length = is.tellg();
             is.seekg(0, is.beg);
             std::vector<char> buffer;
-            buffer.resize(length);
+            buffer.resize(static_cast<size_t>(length));
             is.read(&buffer[0], length);
             is.close();
             str_ = std::string(buffer.begin(), buffer.end());
